operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -107,58 +110,30 @@ private String getLockName(String namespace, String name) {
      */
     @Override
     public final Future<Void> reconcile(Reconciliation reconciliation) {
-        Future<Void> handler = Future.future();
         String namespace = reconciliation.namespace();
         String name = reconciliation.name();
-        final String lockName = getLockName(namespace, name);
-        vertx.sharedData().getLockWithTimeout(lockName, LOCK_TIMEOUT_MS, res -> {
-            if (res.succeeded()) {
-                log.debug("{}: Lock {} acquired", reconciliation, lockName);
-                Lock lock = res.result();
-
-                try {
-                    T cr = resourceOperator.get(namespace, name);
-                    if (cr != null) {
-                        validate(cr);
-                        log.info("{}: {} {} should be created or updated", reconciliation, kind, name);
-
-                        createOrUpdate(reconciliation, cr).setHandler(createResult -> {
-                            lock.release();
-                            log.debug("{}: Lock {} released", reconciliation, lockName);
-                            if (createResult.failed()) {
-                                log.error("{}: createOrUpdate failed", reconciliation, createResult.cause());
-                            } else {
-                                handler.handle(createResult);
-                            }
-                        });
+        Future<Void> handler = withLock(reconciliation, LOCK_TIMEOUT_MS, () -> {
```

Comment:
I did this to avoid needing to to reconcile the whole KafkaConnect when only a Connector has changed. I.e. with the lock held I can just call `reconcileConnectors()` instead of a full on `reconcile()`.

I don't think the old code was as well tested as you imply, because in the course of the refactoring I found an existing bug. And actually if you compare the code of `withLock()` it's actually cleaner that what existed before. Having a separate method for the locking gives us the opportunity to write tests just for the locking, though I've not done that in this PR yet.

Reply:
Well, I didn't implied it was well tested before. But we managed to catch a lot of issues over the time. But yes, if it had some reason than OK.